src/EditorFeatures/CSharpTest/Diagnostics/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -3790,5 +3790,22 @@ private static void RequiresCondition<TException>(bool condition, string message
     }
 }");
         }
+
+        [WorkItem(8111, "https://github.com/dotnet/roslyn/issues/8111")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
+        public async Task DontRemoveCastThatChangesShapeOfAnonymousTypeObject()
+        {
+            await TestMissingAsync(
+            @"
+class Program
+{
+    static void Main(string[] args)
+    {
+        object thing = new { shouldBeAnInt = [|(int)Directions.South|] };
+    }
+    public enum Directions { North, East, South, West }
+}
+");
+        }
```

Comment:
It's be good to add a test to ensure that it still reports redundant identity casts in this scenario.